### PR TITLE
NotificationJob

### LINKED ISSUE #173: Email notifications on old unreconciled txns

### PR DESCRIPTION
**Pull request by sprucely**
Sends notification of failed (Rejected or stale Processing) transactions to users with a team role of Admin, cc'ing users with role of Manager. Provides linkback to the failed txn report for given team.

The job will run nightly, so we may not want to merge until we have the ability to mark transactions as cancelled.

resolves #173

### CODE REVIEW

**Comment by jSylvestre** [comment]
@sprucely might want to think about issue #323 with this.

File: Sloth.Jobs.Notifications/Sloth.Jobs.Notifications.csproj [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
+    <UserSecretsId>738f155f-718e-45d3-aceb-cb0b5d4a04f8</UserSecretsId>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Content Include="appsettings.json">
+      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
```
Comment (jSylvestre) [inline Sloth.Jobs.Notifications/Sloth.Jobs.Notifications.csproj:12]: Just noticing this is different from the other job I looked at. Maybe better than PreserveNewest.

**Review by jSylvestre — APPROVED** [review]
Didn't test, but looks ok.

File: Sloth.Core/Domain/JobRecord.cs [hunk 2]
```diff
@@ -34,6 +34,11 @@ public JobRecord()
 
         public void SetDetails(object details)
         {
+            if (details is string stringDetails)
+            {
+                Details = stringDetails;
+                return;
```
Comment (srkirkland) [inline Sloth.Core/Domain/JobRecord.cs:40]: I'd prefer just an if/else here, the early return makes sense but isn't super easy to understand when skimming

File: Sloth.Core/Extensions/StringExtensions.cs [hunk 3]
```diff
@@ -59,8 +59,15 @@ public static string SafeRegexRemove(this string value, string regEx = @"[^0-9a-
...
         public static string JsonPrettify(this string json)
         {
-            using var jDoc = JsonDocument.Parse(json);
-            return JsonSerializer.Serialize(jDoc, new JsonSerializerOptions { WriteIndented = true });
+            try
+            {
+                using var jDoc = JsonDocument.Parse(json);
+                return JsonSerializer.Serialize(jDoc, new JsonSerializerOptions { WriteIndented = true });
+            }
+            catch (Exception)
```
Comment (srkirkland) [inline Sloth.Core/Extensions/StringExtensions.cs:67]: just do `try{ .. } catch { }` if you aren't going to use the exception.  You don't need the param arg.

File: Sloth.Core/Migrations/20230303195053_TxnLastModified.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        {
+            migrationBuilder.AddColumn<DateTime>(
+                name: "LastModified",
+                table: "Transactions",
+                type: "datetime2",
+                nullable: false,
+                defaultValueSql: "SYSUTCDATETIME()");
+
+            // update LastModified with max EventDate, falling back to TransactionDate if it has no status events
+            migrationBuilder.Sql(@"
```
Comment (srkirkland) [inline Sloth.Core/Migrations/20230303195053_TxnLastModified.cs:20]: neat

**Review by srkirkland — APPROVED** [review]

**Pull request merged**